state/block.go

```diff
@@ -194,13 +195,30 @@ func (bs *blockState) SetBabeHeader(epoch uint64, slot uint64, blockData babe.Ba
 	return err
 }
 
-// AddBlock will add a block into the DB
-func (bs *blockState) AddBlock(block types.BlockHeader) error {
+// AddBlock will set the latestBlock in blockState DB
+func (bs *blockState) AddBlock(block types.Block) error {
+	blockHeader := *block.Header
+
 	// Set the latest block
-	if block.Number.Cmp(bs.latestBlock.Number) == 1 {
-		bs.latestBlock = block
+	if reflect.DeepEqual(bs.latestBlock, types.BlockHeader{}) {
```

Comment:
maybe we can change blockState.latestBlock into a pointer? comparing to nil seems cleaner

Reply:
yep